tar/tar.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Copyright 2020 The FluxCD contributors. All rights reserved.
+// Adapted from: golang.org/x/build/internal/untar
+
+// Package untar untars a tarball to disk.
+package tar
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type Tar struct {
+	maxUntarSize int
+}
+
+func NewTar() *Tar {
+	return &Tar{
+		maxUntarSize: 1 << (10 * 3), // 1Gb
```

Comment:
Added 1Gb to decrease the likelihood of breaking existing usage, but it feels too large for a default value.
Keen to here opinions on better values here.

Reply:
I would set the default to `0` (no limit) to make it fully backwards compatible, and instead break it on the implementation side. Which makes this an "API addition only" change.

Reply:
Fair point, I took a slight different approach by setting the default `maxUntarSize` to 100Mb for consumers of the new API. But consumers with automatically have no limit when using the old API. PTAL